tabular/src/autogluon/tabular/learner/abstract_learner.py

```diff
@@ -681,9 +681,20 @@ def _remove_nan_label_rows(X, y):
             X = X.loc[y.index]
         return X, y
 
+    def infer_problem_type(self, y: Series, silent=False):
+        problem_type = self._infer_problem_type(y, silent=silent)
+        if self.quantile_levels is not None:
+            if problem_type == REGRESSION:
+                problem_type = QUANTILE
+            else:
+                raise ValueError("autogluon infers this to be classification problem for which quantile_levels "
+                                 "cannot be specified. If it is truly a quantile regression problem, "
+                                 "please specify:problem_type='quantile'")
```

Comment:
this would raise an error if the `problem_type == QUANTILE`

Reply:
It is impossible for it to be quantile because `self._infer_problem_type` never returns QUANTILE. The logic is unchanged from before, just moved around.

Reply:
addressed